lib/teams/index.ts

```diff
@@ -214,6 +226,98 @@ export class ApplicationTeam implements Team {
                 hard: this.teamProps.namespaceHardLimits
             }
         });
+        quotaManifest.node.addDependency(this.namespaceManifest);
+    }
+
+    /**
+     * Sets up secrets
+     * @param clusterInfo
+     */
+    protected setupSecrets(clusterInfo: ClusterInfo) {
+        const secrets = this.teamProps.secrets;
+        if (clusterInfo.provisionedAddOns) {
```

Comment:
We could refactor this if clusterInfo made a class like this:
```
const csiDriver = clusterInfo.getProvisionedAddon(Constants. SECRET_STORE_CSI_DRIVER);
console.assert(csiDriver != null, "your error message");

secret and sa provisioning code
```

The above will reduce the number of `if` and nesting.

Reply:
Done in 5069a9821ee402ca5e8db6a5d7a2310346f3e64e